Sources/ZIPFoundation/Archive+Writing.swift

```diff
@@ -228,15 +228,15 @@ extension Archive {
 
     private func writeUncompressed(size: UInt32, bufferSize: UInt32,
                                    provider: Provider) throws -> (sizeWritten: UInt32, checksum: CRC32) {
-        var position = 0
+        var position: UInt32 = 0
         var sizeWritten = 0
         var checksum = CRC32(0)
         while position < size {
-            let readSize = (Int(size) - position) >= bufferSize ? Int(bufferSize) : (Int(size) - position)
+            let readSize = (size - position) >= bufferSize ? bufferSize : (size - position)
             let entryChunk = try provider(Int(position), Int(readSize))
             checksum = entryChunk.crc32(checksum: checksum)
             sizeWritten += try Data.write(chunk: entryChunk, to: self.archiveFile)
-            position += Int(bufferSize)
+            position += bufferSize
```

Comment:
This introduces an overflow crash. 
If size is `UInt32.max` and `position` is `UInt32`, the last iteration crashes. This is caught by the test suite in `testCreateArchiveAddTooLargeUncompressedEntry`.
I assume you got "binary operator cannot be applied" warnings. Swift 4 seems to implicitly upcast `UInt32` to make it compatible with `Int` here. Maybe you can keep `position` as `Int` and cast the `UInt32` values where necessary to make it compile with older Swift versions.